src/test/localfilesystem_test.cpp

```diff
@@ -166,3 +166,50 @@ TEST(FileSystem, CheckIfPathIsEscaped) {
     ASSERT_TRUE(!ovms::FileSystem::isPathEscaped("/path/..resnet/"));
     ASSERT_TRUE(!ovms::FileSystem::isPathEscaped("/path/resnet../"));
 }
+
+TEST(FileSystem, IsLocalFilesystem) {
+    ASSERT_FALSE(ovms::FileSystem::isLocalFilesystem("s3://"));
+    ASSERT_FALSE(ovms::FileSystem::isLocalFilesystem("gs://"));
+    ASSERT_FALSE(ovms::FileSystem::isLocalFilesystem("azfs://"));
+    ASSERT_FALSE(ovms::FileSystem::isLocalFilesystem("az://"));
+    ASSERT_TRUE(ovms::FileSystem::isLocalFilesystem("nanas3://"));
+    ASSERT_TRUE(ovms::FileSystem::isLocalFilesystem("...gs://"));
+    ASSERT_TRUE(ovms::FileSystem::isLocalFilesystem("/azfs://"));
+    ASSERT_TRUE(ovms::FileSystem::isLocalFilesystem("o_O$az://"));
+    ASSERT_TRUE(ovms::FileSystem::isLocalFilesystem("../"));
+    ASSERT_TRUE(ovms::FileSystem::isLocalFilesystem("/localfilesystem"));
+    ASSERT_TRUE(ovms::FileSystem::isLocalFilesystem("/long/local/filesystem"));
+}
+
+TEST(FileSystem, setRootDirectoryPath) {
+    std::string rootPath = "";
+    std::string givenPath = "/givenpath";
+    auto fs = std::make_shared<ovms::LocalFileSystem>();
```

Comment:
We should not be required to create LocalFileSystem class instance just to call `joinPath`. The function does not use any internal field, therefore it should be static.

What's more, joinPath exists in 2 classes - in LocalFileSystem and in AzureStorageAdapter - both methods are identical (and both non static).